src/wp-admin/includes/image-edit.php

```diff
@@ -826,19 +826,21 @@ function wp_restore_image( $post_id ) {
 				}
 			} elseif ( isset( $meta['width'], $meta['height'] ) ) {
 				$backup_sizes[ "full-$suffix" ] = array(
-					'width'  => $meta['width'],
-					'height' => $meta['height'],
-					'file'   => $parts['basename'],
+					'width'    => $meta['width'],
+					'height'   => $meta['height'],
+					'filesize' => $meta['filesize'],
+					'file'     => $parts['basename'],
 				);
 			}
 		}
 
 		$restored_file = path_join( $parts['dirname'], $data['file'] );
 		$restored      = update_attached_file( $post_id, $restored_file );
 
-		$meta['file']   = _wp_relative_upload_path( $restored_file );
-		$meta['width']  = $data['width'];
-		$meta['height'] = $data['height'];
+		$meta['file']     = _wp_relative_upload_path( $restored_file );
+		$meta['width']    = $data['width'];
+		$meta['height']   = $data['height'];
+		$meta['filesize'] = $data['filesize'];
```

Comment:
This line throws an error if `$data['filesize']` doesn't exist, which will be the case for every file that was edited prior to this fix being applied. 

```suggestion
		if ( isset( $data['filesize'] ) {
			$meta['filesize'] = $data['filesize'];
		}
```